tests/test_modeling_common.py

```diff
@@ -427,6 +430,58 @@ class CopyClass(model_class):
                         max_diff = (model_slow_init.state_dict()[key] - model_fast_init.state_dict()[key]).sum().item()
                     self.assertLessEqual(max_diff, 1e-3, msg=f"{key} not identical")
 
+    def test_fast_init_context_manager(self):
+        # 1. Create a dummy class. Should have buffers as well? To make sure we test __init__
+        class MyClass(PreTrainedModel):
+            config_class = PretrainedConfig
+
+            def __init__(self, config=PretrainedConfig()):
```

Comment:
Wouldn't you rather pass the raw class and instantiate it within the method? Here it'll be instantiated once; if you re-use it you'll get the potential changes done to the class in previous initializations.

Reply:
Yes